src/views/Room.vue

```diff
@@ -19,79 +21,61 @@ import TheNavSmall from '@/components/navigation/TheNavSmall.vue'
 import TheToolPanel from '@/components/TheToolPanel.vue'
 import TheCanvas from '@/components/TheCanvas.vue'
 import TheEntityPanel from '@/components/TheEntityPanel.vue'
-import Component from 'vue-class-component'
-import { Prop, Watch } from 'vue-property-decorator'
-import Vue from 'vue'
-import { namespace, Action } from 'vuex-class'
-import { Namespaces } from '@/store'
+import Component, { mixins } from 'vue-class-component'
+import { Prop } from 'vue-property-decorator'
+import { CanvasElement, VueKonvaStage } from '@/types/Canvas'
+import TheCreateNewTacticOverlay from '@/components/overlays/TheCreateNewTacticOverlay.vue'
+import { StageActions, StageGetters } from '@/store/modules/stage'
+import { CustomStageConfig } from '@/util/PointerEventMapper'
+import { CanvasAction } from '@/store/modules/canvas'
+import { Action, Getter } from 'vuex-class'
 import { EventBus } from '@/event-bus'
-import { VueKonvaStage } from '@/types/Canvas'
-import { AuthenticationGetters, ExtendedJWT } from '@/store/modules/authentication'
-import { Socket } from 'vue-socket.io-extended'
-
-const authNamespace = namespace(Namespaces.AUTH)
+import RoomSocket from '@/mixins/RoomSockets'
 
   @Component({
     name: 'Room',
+    mixins: [RoomSocket],
     components: {
+      TheCreateNewTacticOverlay,
       TheNavLarge,
       TheNavSmall,
       TheToolPanel,
       TheCanvas,
       TheEntityPanel
     }
   })
-export default class extends Vue {
+export default class Room extends mixins(RoomSocket) {
   @Prop() id!: string
-  @authNamespace.Getter(AuthenticationGetters.IS_AUTH) isAuth!: boolean
-  @authNamespace.Getter(AuthenticationGetters.JWT) jwt!: ExtendedJWT
-  @Action('socket/joinRoom') joinRoom!: (id: string) => void
-  @Socket() // --> listens to the event by method name, e.g. `connect`
-  connect () {
-    // eslint-disable-next-line
-    console.log('connection established')
-    this.joinRoom(this.id)
-  }
+  @Action(`canvas/${CanvasAction.SET_CANVAS_ELEMENT}`) setCanvasElements!: (canvasElements: CanvasElement[]) => void
+  @Action(`canvas/${CanvasAction.SET_CANVAS_ELEMENT_HISTORY}`) setCanvasElementsHistory!: (canvasElements: CanvasElement[]) => void
+  @Getter(`stage/${StageGetters.STAGE_CONFIG}`) stageConfig!: CustomStageConfig
```

Comment:
Interface StageConfig is taken by konva which is why i had to name it something else
I tried thinking of other names for it like MapConfig or CanvasConfig, but these names didnt make sence since we already have a Map interface, and the Canvas is not the same as the stage. If you have a better name let me know, but in my opinion i think CustomStageConfig is fine